pkg/connector/handletwit.go

```diff
@@ -0,0 +1,179 @@
+package connector
+
+import (
+	"github.com/rs/zerolog"
+	"maunium.net/go/mautrix/bridgev2"
+	"maunium.net/go/mautrix/bridgev2/networkid"
+	"maunium.net/go/mautrix/bridgev2/simplevent"
+
+	"go.mau.fi/mautrix-twitter/pkg/twittermeow/data/types"
+	"go.mau.fi/mautrix-twitter/pkg/twittermeow/event"
+)
+
+func (tc *TwitterClient) HandleTwitterEvent(rawEvt any) {
+	switch evtData := rawEvt.(type) {
+	case event.XEventMessage:
+		sender := evtData.Sender
+		isFromMe := sender.IDStr == string(tc.userLogin.ID)
+		msgType := bridgev2.RemoteEventMessage
+
+		if evtData.EditCount > 0 {
+			msgType = bridgev2.RemoteEventEdit
+		}
+		tc.connector.br.QueueRemoteEvent(tc.userLogin, &simplevent.Message[*event.XEventMessage]{
+			EventMeta: simplevent.EventMeta{
+				Type: msgType,
+				LogContext: func(c zerolog.Context) zerolog.Context {
+					return c.
+						Str("message_id", evtData.MessageID).
+						Str("sender", sender.IDStr).
+						Str("sender_login", sender.ScreenName).
+						Bool("is_from_me", isFromMe)
+				},
+				PortalKey:    tc.MakePortalKey(evtData.Conversation),
+				CreatePortal: true,
+				Sender: bridgev2.EventSender{
+					IsFromMe:    isFromMe,
+					SenderLogin: networkid.UserLoginID(sender.IDStr),
+					Sender:      networkid.UserID(sender.IDStr),
+				},
+				Timestamp: evtData.CreatedAt,
+			},
+			ID:                 networkid.MessageID(evtData.MessageID),
+			TargetMessage:      networkid.MessageID(evtData.MessageID),
+			Data:               &evtData,
+			ConvertMessageFunc: tc.convertToMatrix,
+			ConvertEditFunc:    tc.convertEditToMatrix,
+		})
+	case event.XEventReaction:
+		reactionRemoteEvent := tc.wrapReaction(evtData)
+		tc.connector.br.QueueRemoteEvent(tc.userLogin, reactionRemoteEvent)
+	case event.XEventConversationRead:
+		// conversation read events are only fired by yourself??
+		// if another user reads your message this is never fired
+		// they use user_updates and last_read_event_id to figure that out with the polling client
+		tc.client.Logger.Info().
+			Str("conversation_id", evtData.Conversation.ConversationID).
+			Str("last_read_event_id", evtData.LastReadEventID).
+			Str("read_at", evtData.ReadAt.String()).
+			Msg("Conversation was read!")
+	case event.XEventConversationCreated:
+		// honestly not sure when this is ever called... ? might be when they initialize the conversation with me?
+		tc.client.Logger.Warn().Any("data", evtData).Msg("XEventConversationCreated")
```

Comment:
This also sounds like something that should be bridged. `ChatResync` events can be used to create portals without triggering anything else

Reply:
never got called for me